pkgs/sdk/server/src/Internal/DataStores/PersistentStoreWrapper.cs

```diff
@@ -185,6 +187,22 @@ public void Init(FullDataSet<ItemDescriptor> items)
             }
         }
 
+        public async Task<ItemDescriptor?> GetAsync(DataKind kind, string key, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var ret = _itemCache is null ? await GetAndDeserializeItemAsync(kind, key, cancellationToken) :
+                    _itemCache.Get(new CacheKey(kind, key));
```

Comment:
In high-volume uses we would mostly expect sync cache hits, which makes this a good candidate for `ValueTask<>`.